semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/package.scala

```diff
@@ -1,18 +1,37 @@
 package scala.meta.internal
 
-import java.nio.file._
+import java.nio.file.Files
 import scala.meta.io.AbsolutePath
 import scala.tools.asm._
 import scala.tools.asm.ClassReader._
 import scala.tools.asm.tree._
 
 package object metacp {
+  private val SKIP_ALL = SKIP_CODE | SKIP_DEBUG | SKIP_FRAMES
+
+  implicit class XtensionAsmBytes(entry: ReadableClasspathEntry) {
+    def hasScalaSig: Boolean = {
+      val visitor = new HasScalaSigVisitor
```

Comment:
Let's use `toClassNode` here instead of rolling our own visitor. This will get rid of a class (`HasScalaSigVisitor`) that's not used anywhere else.

Reply:
Are you OK with keeping it? I think it's fine to write custom visitor